src/core/common/Breadcrumbs.js

```diff
@@ -0,0 +1,24 @@
+import React from "react";
+
+import { withStyles } from '@material-ui/core/styles';
+import Typography from '@material-ui/core/Typography';
+
+import styles from "../styles";
```

Comment:
Using a global stylesheet really feels weird here; the advantage of components is to colocate markup, styles, and behaviors. Could you explain why you chose to mutualize styles?

Reply:
You should use solution 1., and material-ui's theming, see https://material-ui.com/customization/themes/

Reply:
I tried to used theming form Material-UI, like:

<MuiThemeProvider theme={theme}>
// My component
</MuiThemeProveder>

where 

theme = createMuiTheme({
   overrides: {
       // Stylesheet
   }
});

But if I understand correctly, MuiThemeProvider works only with Material-ui elements. And for this reason we can't use it for simple <div></div>.

I did following (by the first way):
1) I moved all styles to the components, where they are used;
2) If theme styles aren't necessary - add them to component by **_withStyles()_**
3) If theme styles is necessary - merge Core and Theme styles.
4) Theme styles are located in /version/styles directory (each style in the separate file).
5) Styles storage in core part was removed

Can we use this solution?

Reply:
In my mind using material-ui theming means doing something like:

```jsx
const styles = theme => ({
    root: {
        color: theme.palette.primary,
    }
})

const MyComponent = ({ classes, ...rest }) => (
    <div className={classes.root}>
        ...
    </div>
) 

export withStyles(styles)(MyComponent);
```

That way the color of the div is dependent on the theme. Yes, that implies that you create custom components when you want to add style to a div.

Reply:
you shouldn't have to merge core and Theme styles by hand